src/modules/job-exec/job-exec.c

```diff
@@ -1218,6 +1218,42 @@ static void critical_ranks_cb (flux_t *h,
         flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
 }
 
+static void expiration_cb (flux_t *h,
+                           flux_msg_handler_t *mh,
+                           const flux_msg_t *msg,
+                           void *arg)
+{
+    struct job_exec_ctx *ctx = arg;
+    flux_jobid_t id;
+    double expiration;
+    struct jobinfo *job;
+
+    if (flux_request_unpack (msg,
+                             NULL,
+                             "{s:I s:F}",
+                             "id", &id,
+                             "expiration", &expiration) < 0)
+        goto error;
+
+     if (!(job = zhashx_lookup (ctx->jobs, &id))) {
+        errno = ENOENT;
+        goto error;
+    }
+    resource_set_update_expiration (job->R, expiration);
+    (void) jobinfo_set_expiration (job);
```

Comment:
`jobinfo_set_expiration()` has lots of nice error paths - any reason not to test the return value and `goto error` if one of them catches something?

Reply:
Yes, that makes sense. In the early version this was a noresponse RPC and this is left over from that. I'll fix.